drivers/virtualbox/virtualbox.go

```diff
@@ -572,6 +573,44 @@ func (d *Driver) generateDiskImage(size int) error {
 	return createDiskImage(d.diskPath(), size, raw)
 }
 
+func (d *Driver) setupHostOnlyNetwork(machineName string) error {
+	ip, network, err := net.ParseCIDR(d.HostOnlyCIDR)
+	nAddr := network.IP.To4()
+
+	var dhcpAddr net.IP
+	// select pseudo-random DHCP addr; make sure not to clash with the host
+	for {
```

Comment:
Could we maybe move this code into a function, say, `getAvailableDHCPAddr`,  which returns `dhcpAddr`?